src/TemplateContainer.tsx

```diff
@@ -1,52 +1,53 @@
 import * as React from 'react';
+import type { DragHandleProps, RenderTemplate } from '.';
 
 interface Props<I, C, T> {
   item: I;
-  template: new (props: any, context?: any) => T;
+  renderTemplate: RenderTemplate<I, C, T>;
   itemSelected: number;
   anySelected: number;
-  dragHandleProps: object;
-  commonProps: C;
+  dragHandleProps: DragHandleProps;
+  commonProps?: C;
 }
 
 export default class TemplateContainer<
   I,
   C,
-  T extends React.Component<any, any>
+  T extends React.Component<Props<I, C, T>>
 > extends React.Component<Props<I, C, T>> {
-  private _template: T | undefined;
-  private readonly _templateSetter = (cmp: any) => {
-    this._template = cmp;
-  };
+  #template = React.createRef<T>();
 
-  public shouldComponentUpdate(nextProps: Props<I, C, T>): boolean {
+  shouldComponentUpdate(nextProps: Props<I, C, T>): boolean {
     return (
       this.props.anySelected !== nextProps.anySelected ||
       this.props.itemSelected !== nextProps.itemSelected ||
       this.props.item !== nextProps.item ||
-      this.props.template !== nextProps.template ||
+      this.props.renderTemplate !== nextProps.renderTemplate ||
       this.props.commonProps !== nextProps.commonProps
     );
   }
 
-  public getTemplate(): T {
-    return this._template!;
+  get template() {
+    return this.#template;
```

Comment:
Let's be explicit about the type signature here.

It looks like this function changed to returning the ref object instead of `T`. Let's change it back to returning `T` by doing `return this.#template.current;`.

Reply:
[c19b6c9](https://github.com/StreakYC/react-draggable-list/pull/62/commits/c19b6c9aef71990b5bde30ad2cafb9952b197d8b)

I think the non-null assertion isn't helpful here, because we ultimately do have to make the non-null check in `index.tsx`.

Reply:
Checking the `T` return type checking is handled at the top level in [0e302f8](https://github.com/StreakYC/react-draggable-list/pull/62/commits/0e302f89df55444b546bc56818cbc0688ec230bf).